src/Compilers/CSharp/Portable/Symbols/Synthesized/InjectedNonNullTypesAttributeSymbol.cs

```diff
@@ -0,0 +1,173 @@
+            }
+        }
+
+        internal override bool RequiresCompletion
+            => true;
+
+        internal override bool HasComplete(CompletionPart part)
+            => _state.HasComplete(part);
+
+        internal override void ForceComplete(SourceLocation locationOpt, CancellationToken cancellationToken)
+        {
+            while (true)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
+                var incompletePart = _state.NextIncompletePart;
+                switch (incompletePart)
+                {
+                    case CompletionPart.Attributes:
+                        GetAttributes();
+                        break;
+
+                    case CompletionPart.None:
+                        return;
+
+                    default:
+                        // any other values are completion parts intended for other kinds of symbols
+                        _state.NotePartComplete(CompletionPart.All & ~CompletionPart.InjectedSymbolAll);
+                        break;
+                }
+
+                _state.SpinWaitComplete(incompletePart, cancellationToken);
+            }
+
+            throw ExceptionUtilities.Unreachable;
+        }
+
+        public override ImmutableArray<CSharpAttributeData> GetAttributes()
+        {
+            var attributes = _lazyCustomAttributes;
+            if (!attributes.IsDefault)
+            {
+                return attributes;
+            }
+
+            if (ImmutableInterlocked.InterlockedInitialize(ref _lazyCustomAttributes, MakeAttributes()))
+            {
+                var completed = _state.NotePartComplete(CompletionPart.Attributes);
+                Debug.Assert(completed);
+            }
+
+            return _lazyCustomAttributes;
+        }
+
+        /// <summary>
+        /// Adds an `[AttributeUsage(AttributeTargets.Class | ...)]` (if possible) and captures any diagnostics in the process.
+        /// </summary>
+        private ImmutableArray<CSharpAttributeData> MakeAttributes()
+        {
+            var ctor = (MethodSymbol)DeclaringCompilation.GetWellKnownTypeMember(WellKnownMember.System_AttributeUsageAttribute__ctor);
+            if (ctor is null)
+            {
+                // member is missing
+                var memberDescriptor = WellKnownMembers.GetDescriptor(WellKnownMember.System_AttributeUsageAttribute__ctor);
+                var diagnosticInfo = new CSDiagnosticInfo(ErrorCode.ERR_MissingPredefinedMember, memberDescriptor.DeclaringTypeMetadataName, memberDescriptor.Name);
+                Diagnostics.Add(diagnosticInfo, Location.None);
+
+                return ImmutableArray<CSharpAttributeData>.Empty;
+            }
+
+            Binder.ReportUseSiteDiagnostics(ctor, Diagnostics, Location.None);
+
+            NamedTypeSymbol attributeTargets = DeclaringCompilation.GetWellKnownType(WellKnownType.System_AttributeTargets);
```

Comment:
>DeclaringCompilation.GetWellKnownType(WellKnownType.System_AttributeTargets); [](start = 47, length = 77)

Should we look for the right constructor instead?

Reply:
I didn't follow. 
To create a `TypedConstant` in an `AttributeData`, we only need the type and a value, and we can get the type we need from `GetWellKnownType`.
This pattern occurs in multiple places in the compiler, for instance `SynthesizeDecimalConstantAttribute`.

Reply:
> To create a TypedConstant in an AttributeData, we only need the type and a value, and we can get the type we need from GetWellKnownType.

I see that we are looking for the constructor right before this line.

---
In reply to: [215092640](https://github.com/dotnet/roslyn/pull/29180#discussion_r215092640) [](ancestors = 215092640)